omod/src/main/java/org/openmrs/module/owa/activator/OwaActivator.java

```diff
@@ -76,6 +78,30 @@ public void contextRefreshed() {
 		catch (IOException e) {
 			log.error(e);
 		}
+
+		/**
+		 * Check if there is any '.zip' file in OWA app directory and deploy them
+		 * this app manager is out of Spring context, so apps have to be reloaded here
+		 */
+		AppManager appManager = new DefaultAppManager();
+		appManager.reloadApps();
+		File owaDir = new File(Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_FOLDER_PATH));
+		File[] files = owaDir.listFiles();
+		if (files != null) {
+			for (File file : files) {
+				if (file.getName().endsWith(".zip")) {
+					try {
+						appManager.installApp(file, file.getName(), Context.getAdministrationService().getGlobalProperty(AppManager.KEY_APP_BASE_URL));
+						log.info("Deploying OWA from: "+file.getName()+"...");
```

Comment:
Shouldn't this log be before installApp? Or at least changed its text to Deployed. :)